src/manifolds/Grassmann.jl

```diff
@@ -380,6 +380,23 @@ end
 
 show(io::IO, ::Grassmann{n,k,𝔽}) where {n,k,𝔽} = print(io, "Grassmann($(n), $(k), $(𝔽))")
 
+"""
+    uniform_distribution(M::Grassmann{n,k,ℝ}, p)
+
+Uniform distribution on given (real-valued) [`Grassmann`](@ref) `M`.
+Generated points will be of similar type as `p`.
+
+The implementation is based on Section 2.5.1 in [^Chikuse2003];
+see also Theorem 2.2.2(iii) in [^Chikuse2003].
+
+[^Chikuse2003]:
+    > Y. Chikuse: "Statistics on Special Manifolds", Springer New York, 2003,
+    > doi: [10.1007/978-0-387-21540-2](https://doi.org/10.1007/978-0-387-21540-2).
+"""
+function uniform_distribution(M::Grassmann{n,k,ℝ}, p) where {n,k}
+    return uniform_distribution(Stiefel(n, k, ℝ), p)
```

Comment:
If you want to re-use the uniform distribution implementation for `Stiefel`, you should make a helper function that is called from both methods of `uniform_distribution`. Otherwise the distribution returned for `Grassmann` will have wrong manifold set.

Reply:
Good point! I've implemented a separate `uniform_distribution` for `Grassmann`. The code overlaps but I think this is probably the conceptually simpler approach. :)